katharsis-core/src/main/java/io/katharsis/core/internal/resource/ResourceFieldImpl.java

```diff
@@ -143,8 +146,17 @@ public int hashCode() {
 	public ResourceInformation getParentResourceInformation() {
 		return parentResourceInformation;
 	}
-
+	
+	@Override
+	public ResourceFieldAccessor getAccessor(){
+		if(accessor == null){
```

Comment:
If it' no allowed on `get` then it shouldn't get allowed on `set`. Based on the number of constructor args and what you're trying perform, I think a builder pattern is best. Then, you can check the `.build()` and verify things are in order before returning an instance. Immutability is generally not advisable and alas, Java doesn't make this it's default state. 

FYI: Lombok can give you a builder with 1 annotation